electron/gog/library.ts

```diff
@@ -0,0 +1,587 @@
+import axios, { AxiosError, AxiosResponse } from 'axios'
+import Store from 'electron-store'
+import { GOGUser } from './user'
+import {
+  GOGLoginData,
+  GOGGameInfo,
+  GameInfo,
+  InstallInfo,
+  InstalledInfo,
+  GOGImportData
+} from '../types'
+import { join } from 'node:path'
+import { existsSync, readFileSync } from 'graceful-fs'
+import prettyBytes from 'pretty-bytes'
+import { logError, logInfo, LogPrefix, logWarning } from '../logger/logger'
+import { execAsync } from '../utils'
+import { fallBackImage, gogdlBin, isMac } from '../constants'
+
+const userStore = new Store({
+  cwd: 'gog_store'
+})
+const apiInfoCache = new Store({ cwd: 'gog_store', name: 'api_info_cache' })
+const libraryStore = new Store({ cwd: 'gog_store', name: 'library' })
+const installedGamesStore = new Store({
+  cwd: 'gog_store',
+  name: 'installed'
+})
+
+export class GOGLibrary {
+  private static globalInstance: GOGLibrary = null
+  private library: Map<string, null | GameInfo> = new Map()
+  private installedGames: Map<string, null | InstalledInfo> = new Map()
+
+  private constructor() {
+    this.refreshInstalled()
+  }
+
+  public async sync() {
+    if (!GOGUser.isLoggedIn()) {
+      return
+    }
+    if (GOGUser.isTokenExpired()) {
+      await GOGUser.refreshToken()
+    }
+    // This gets games ibrary
+    // Handles multiple pages
+    this.refreshInstalled()
+    const credentials: GOGLoginData = userStore.get(
+      'credentials'
+    ) as GOGLoginData
+    const headers = { Authorization: 'Bearer ' + credentials.access_token }
+    logInfo('Getting GOG library', LogPrefix.Gog)
+    let gameApiArray: Array<GOGGameInfo> = []
+    const games = await axios
+      .get(
+        'https://embed.gog.com/account/getFilteredProducts?mediaType=1&sortBy=title',
+        { headers }
+      )
+      .catch((e: AxiosError) => {
+        logError(
+          ['There was an error getting games library data', e.message],
+          LogPrefix.Gog
+        )
+        return null
```

Comment:
I believe we should return an empty array here, if we return `null` the library component might break.